Add mailman2sympa migration script with my additions

LAKostis opened the pull request:
mailman2sympa 0.7, Konstantin Lepikhov - Nov 2020

+ rewrote configuration parsing (use python unpickle)
+ import list info (if exists)
+ don't import if list host_name doesn't match
+ handle unicode names for users/subscribers

LAKostis committed:
Add mailman2sympa migration script with my additions

mailman2sympa 0.7, Konstantin Lepikhov - Nov 2020

+ rewrote configuration parsing (use python unpickle)
+ import list info (if exists)
+ don't import if list host_name doesn't match
+ handle unicode names for users/subscribers

ikedas commented:
@LAKostis , thank you for adding migration from Mailman!

There are a few points not very essential:
  - `awk/.subscribers.awk.swp` seems not necessary.
  - ~~`README.pm`~~`README.md` would be better for the name of redme file.

LAKostis committed:
mailman2sympa.sh: add notes for password

- show notes about `upgrade_sympa_password.pl` run after users import
  (as users password imported as clear text and need to converted to
  hashes).

LAKostis committed:
Changes: update

LAKostis committed:
BUGS: update

LAKostis committed:
README: convert to md

LAKostis committed:
TODO: update

LAKostis committed:
awk: drop vim leftovers

LAKostis committed:
scripts: remove mm2s_info

- jq does the better job.

LAKostis committed:
docs: update password upgrade command

LAKostis committed:
README: add link to mailman2sympa

LAKostis commented:
> @LAKostis , thank you for adding migration from Mailman!
> 
> There are a few points not very essential:
> 
>     * `awk/.subscribers.awk.swp` seems not necessary.
> 
>     * ~`README.pm`~`README.md` would be better for the name of redme file.

done!

taggart commented:
@LAKostis you made this PR quite a while ago(2021), but I see you have also made some additional updates since then(2023). This important tool is still missing from the sympa-contrib repo and I would like to be able to point people at it. Is this still ok to merge as is? Anything that needs to change?

BTW the older 0.6 version is still listed with tgz on the older sympa website [here](https://www.sympa.org/contribs/migration_and_archives/index#mailman2sympa) but it's not mentioned anywhere I could find on sympa.community.

LAKostis commented:
@taggart Yes, It's been a while since this MR was offered and I have an updated [version already](https://github.com/LAKostis/mailman2sympa) with additions from other people. I can revisit this MR or someone can put a link to my repo instead. I don't have any mailman installations now, so maybe just merge it here will be the best option.
